src/GrumPHP/Task/PHPUnitCoverageTask.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Configuration\GrumPHP;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use GrumPHP\Task\TaskInterface;
+use InvalidArgumentException;
+use SimpleXMLElement;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class PHPUnitCoverageTask implements TaskInterface
+{
+    /**
+     * @var GrumPHP
+     */
+    protected $grumPHP;
+
+    /**
+     * @param GrumPHP $grumPHP
+     */
+    public function __construct(GrumPHP $grumPHP)
+    {
+        $this->grumPHP = $grumPHP;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfiguration()
+    {
+        $configured = $this->grumPHP->getTaskConfiguration($this->getName());
+
+        return $this->getConfigurableOptions()->resolve($configured);
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'phpunit_coverage';
+    }
+
+    /**
+     * @return OptionsResolver
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'clover_file' => null,
+            'level' => 100,
+        ]);
+
+        $resolver->addAllowedTypes('clover_file', ['string']);
+        $resolver->addAllowedTypes('level', ['int', 'float']);
+
+        return $resolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canRunInContext(ContextInterface $context)
+    {
+        return ($context instanceof GitPreCommitContext || $context instanceof RunContext);
+    }
+
+    public function run(ContextInterface $context)
+    {
+        $percentage = round(min(100, max(0, (float) $this->getConfiguration()['level'])), 2);
+
+        if (!file_exists($this->getConfiguration()['clover_file'])) {
+            throw new InvalidArgumentException('Invalid input file provided');
+        }
+
+        if (!$percentage) {
+            throw new InvalidArgumentException('An integer checked percentage must be given as second parameter');
+        }
+
+        $xml             = new SimpleXMLElement(file_get_contents($this->getConfiguration()['clover_file']));
+        $metrics         = $xml->xpath('//metrics');
```

Comment:
Note: you can probably reduce this to two xpath expressions ( no need for looping )